src/main/com/yetanalytics/datasim/xapi/statement.clj

```diff
@@ -198,10 +208,14 @@
         ;; addition of `:spec` key to 0 or more `template-rules`
         template-rules! (ext/derive-generation-hint profiles rng template-rules)]
     (with-meta
-      (rule/apply-rules-gen base-stmt template-rules! :seed (random/rand-long rng))
+      (let [stmt (rule/apply-rules-gen base-stmt
+                                       template-rules!
+                                       :seed (random/rand-long rng))]
+        ;; Need to override object again, in case object is changed
+        ;; by the Template rules.
+        (if obj-override (assoc stmt "object" obj-override) stmt))
```

Comment:
Question - with this override after rules application do we even need the override before? What's the behavioral difference if we kill the first one?

Reply:
To me, it didn't make sense for an object to be generated via the usual means if will get overridden anyways. So really the first assoc w/ the overriding object acts as a placeholder. Basically an optimization (albeit a hacky one).

Reply:
Now that I think of it, maybe using an empty map as the placeholder object would make it more clear?

Reply:
Ah, so it prevents pure object generation from happening. That's fine I agree lets leave it.